test/buy-tshirt.spec.ts

```diff
@@ -3,30 +3,52 @@ import { MenuContentPage, ProductListPage, ProductAddedModalPage,
   SummaryStepPage, SignInStepPage, AddressStepPage,
   ShippingStepPage, OrderSummaryPage, BankPaymentPage, PaymentStepPage} from '../src/page';
 
-describe('Buy a t-shirt', () => {
-  const menuContentPage: MenuContentPage = new MenuContentPage();
-  const productListPage: ProductListPage = new ProductListPage();
-  const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
-  const summaryStepPage: SummaryStepPage = new SummaryStepPage();
-  const signInStepPage: SignInStepPage = new SignInStepPage();
-  const addressStepPage: AddressStepPage = new AddressStepPage();
-  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
-  const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
-  const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
-  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
-  it('then should be bought a t-shirt', async () => {
+describe('This must open the shopping website in the browser', () => {
+  beforeAll(async() => {
     await browser.get('http://automationpractice.com/');
-    await menuContentPage.goToTShirtMenu();
-    await productListPage.addToCart();
-    await productAddedModalPage.clickModalBtn();
-    await summaryStepPage.clickCheckOut();
-    await signInStepPage.performLogin();
-    await addressStepPage.clickCheckOut();
-    await shippingStepPage.acceptTermsOfService();
-    await shippingStepPage.clickCheckOut();
-    await paymentStepPage.clickPayByBank();
-    await bankPaymentPage.clickConfirmOrder();
-    await expect(orderSummaryPage.getOrderStatus())
-    .toBe('Your order on My Store is complete.');
+  });
+  describe('Then it must do the process when i want to buy a t-shirt', () => {
+    beforeAll(async() => {
+      const menuContentPage: MenuContentPage = new MenuContentPage();
+      const productListPage: ProductListPage = new ProductListPage();
+      const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
+      const summaryStepPage: SummaryStepPage = new SummaryStepPage();
+
+      await menuContentPage.goToTShirtMenu();
+      await productListPage.addToCart();
+      await productAddedModalPage.clickModalBtn();
+      await summaryStepPage.clickCheckOut();
+    });
+
+    describe('Then it must login into the website', () => {
+      beforeAll(async() => {
+        const signInStepPage: SignInStepPage = new SignInStepPage();
+        await signInStepPage.performLogin();
+      });
+
+      describe('After that, it must select the default address', () => {
```

Comment:
It should follow a Given.. When... Then... structure...Please improve your description